fix for character creation AFK

Vorshim92 opened the pull request:
An idea came to me to fix the AFK problem during character creation! As you can see, there's the OnCharacterDeath event, which is perfect. I remove the other events upon death, and they are reinitialized when the character is created. Consider that I replaced the OnConnected event with OnCreatePlayer because the latter is not an event that occurs when the player is created during character creation, but it's an event that's triggered when the IsoPlayer is generated during login. So, in theory, it should work smoothly both at login with an already created character and after character creation post-death. If you have a way to test the build this way—otherwise, I'll do it shortly—but I'm quite sure it will work.

Vorshim92 committed:
fix for character creation AFK

Vorshim92 commented:
ok! it's working perfectly.
login with my pg
LOG  : General     , 1726316428043> 0> [AwayFromZomboidfix]   AwayFromZomboid: 1.0.0 initialized.
death
LOG  : General     , 1726316896757> 672.380.868> [AwayFromZomboidfix]   AwayFromZomboid: 1.0.0 Character death.
new pg
LOG  : General     , 1726317155574> 672.639.708> [AwayFromZomboidfix]   AwayFromZomboid: 1.0.0 initialized.

i set 2 minutes for the afk time and it works. when I'm creating the new pg no afk. ;)

Vorshim92 committed:
fix halonote time

MaxWasUnavailable commented:
I'm in the middle of a move to a new apartment, but this sounds promising!

I'll review this asap! Worst case a week or two, depending on when we get internet sorted out so I can pull on my desktop and test locally!

Vorshim92 commented:
> I'm in the middle of a move to a new apartment, but this sounds promising!
> 
> I'll review this asap! Worst case a week or two, depending on when we get internet sorted out so I can pull on my desktop and test locally!

Hey there!
I hope everything is going well with your move, and I wish you a speedy finish. Regarding the mod, of course, whenever you’re able to sit at your computer—I understand it might be a bit tricky right now! No rush xD. As for the testing, I went ahead and checked it myself, and everything is working perfectly. When you die, the event disables everything, and then when creating a new character, everything reactivates. This prevents being kicked while creating the character. Additionally, I changed the function you had for displaying the text above the player’s head. I replaced it with a customizable version, where you can also adjust how long the text stays visible. I synced it with the same time setting as the sandbox, so the text won’t disappear, and you’ll always know if you’re AFK or not!
Hope you enjoy it! Bye!!!

Vorshim92 committed:
changed and fix events for a better result.

Vorshim92 commented:
Hey! How's it going? Still caught up in the chaos of the move, I imagine! I definitely don't want to rush you, just sending a small reminder because I have a lot of players on my server who keep getting kicked during character creation! Thanks and sorry again!

MaxWasUnavailable commented:
We just got internet installed yesterday (finally), so I'll have a look at this in the afternoon! Apologies for the wait.

MaxWasUnavailable commented:
> We just got internet installed yesterday (finally), so I'll have a look at this in the afternoon! Apologies for the wait.

Quick update that I moved my desktop over to the apartment, so I have my dev environment now. Btw, if you need me / want to remind me (which is totally fine), you can reach me on Discord under `maxwasunavailable`

MaxWasUnavailable commented on Contents/mods/AwayFromZomboid/media/lua/client/AwayFromZomboid/AwayFromZomboid.lua:
```diff
@@ -307,7 +312,7 @@ end
 AwayFromZomboid.manualAFKHook = function(chatMessage, tabId)
     if AwayFromZomboid.getAllowManualAFK() then
-        if chatMessage:getText() == "afk" and chatMessage:getAuthor() == getPlayer():getUsername() then
+        if (chatMessage:getText() == "AFK" or chatMessage:getText() == "afk" or chatMessage:getText() == "Afk") and chatMessage:getAuthor() == getPlayer():getUsername() then
```
This could probably be turned into `chatMessage:getText().lower() == "afk"` tbh? I assume/hope PZ's LUA flavour supports that.

Vorshim92 replied:
I think is "string.lower(chatMessage:getText())" or something like this

MaxWasUnavailable commented on Contents/mods/AwayFromZomboid/media/lua/client/AwayFromZomboid/AwayFromZomboid.lua:
```diff
@@ -353,4 +358,14 @@ end
 
-Events.OnConnected.Add(AwayFromZomboid.init)
\ No newline at end of file
+Events.OnCreatePlayer.Add(AwayFromZomboid.init)
```
Just to confirm, this is also ran when a player already exists, right? Not only during initial character creation?

Vorshim92 replied:
Exactly

MaxWasUnavailable commented on Contents/mods/AwayFromZomboid/media/lua/client/AwayFromZomboid/AwayFromZomboid.lua:
```diff
@@ -353,4 +358,14 @@ end
\ No newline at end of file
+Events.OnCreatePlayer.Add(AwayFromZomboid.init)
+Events.OnPlayerDeath.Add(function (player)
+    if getPlayer():isDead() then
```
This can probably be removed, and the OnCreatePlayer hook can be unhooked at the end of the init function. For code style cohesion, I'd also prefer not to have function definitions inside of event hooking.

MaxWasUnavailable commented:
Good stuff! Just had some minor nitpicks & questions

Vorshim92 committed:
fix manualAFKHook and sendChatNotification behaviour

MaxWasUnavailable commented:
Merging into a separate branch so I can revise some things (e.g. adding some additional config flags) before merging into master. Thanks!
